peadoc.js

```diff
@@ -129,58 +129,59 @@ function isSpace(c) {
 	return /\s/.test(c);
 }
 
-function update() {
-	let html = "";
-	let closing = "";
+function update(raw) {
+	let rendered = document.getElementById('pretty');
+	rendered.replaceChildren();
+	let current = [rendered];
 	let prev = EMPTY;
-	const src = document.getElementById('raw').innerText;
-	for (const item of getItems(src)) {
-		if (closing != "" && item.type != prev) {
-			html += closing;
-		}
-		if (item.type == HEADING) {
-			html += "<h1>\n" + escapeLines(item.lines) + "</h1>\n";
-			closing = "";
-		} else if (item.type == TEXT) {
-			html += "<p>\n" + escapeLines(item.lines) + "</p>\n";
-			closing = "";
-		} else if (item.type == PRE) {
-			html += "<pre>\n" + escapeLines(item.lines) + "</pre>\n";
-			closing = "";
-		} else if (item.type == OITEM || item.type == UITEM) {
+	for (const item of getItems(raw.value)) {
+		if (item.type != OITEM && item.type != UITEM) {
+			if (prev == OITEM || prev == UITEM) {
```

Comment:
I think instead of having the variable prev, we could use `current.at(-1).tag` (or whatever gets the element tag). It should be either the `<div>` for `<div id=pretty>` or it should be a `<ul>` or `<ol>`, I believe.